src/libraries/System.Private.Xml/src/System/Xml/XmlConvert.cs

```diff
@@ -107,6 +104,13 @@ public class XmlConvert
                 matchPos = m.Index;
             }
 
+            ValueStringBuilder bufBld =
+                matchPos == -1
+                    ? default
+                    : length < 50
```

Comment:
would be good to create constant for this, also I'd increase this to 128 or 256 which are the two common numbers we use in other places